### ISSUE TITLE
Booleans in conditional statements become strings

### ISSUE THREAD

**Issue by raackley** [hashicorp/terraform#19618]
### Terraform Version
```
Terraform v0.11.10
```

### Expected Behavior
We are using a boolean in a conditional statement like the following:

```
test = "${terraform.env == "green" ? false : true}"
```
When this variable is rendered into a template, it becomes a string value.

When we use a boolean without the conditional statement, the variable gets rendered as a 1 in our template, which is what we expect.

### Actual Behavior
```
test = "${terraform.env == "green" ? false : true}"
```
Renders to 'true'

```
test = true
```
Renders to '1' - expected behavior

**Comment by apparentlymart** [comment]
Hi @raackley! Sorry for the inconsistent behavior and thanks for reporting it.

It's actually the `1` response that was incorrect here, and that has been fixed in master ready to be included in the forthcoming 0.12 release. To be specific, there are two changes related to this:

* Individual expressions can evaluate to bool values _without_ converting to strings. In 0.11 and prior, each separate primitive argument value was always converted to a string before returning.
* The string conversion of `true` is always `"true"`, and likewise the string `"true"` converts to boolean as `true`.

If you need the result to be numeric, it must be written as:

```hcl
test = "${terraform.env == "green" ? 0 : 1}"
```

**Comment by mildwonkey** [comment]
Hi! Since there's been no activity in a few weeks, and there is both a current workaround and a fix in 0.12, I will close this issue. Thank you!
